### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by Maximca2**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,53 +1,16 @@
...
-import NotFound from "./pages/Not found/Not found";
-import { ROOT_TO_MAINPAGE,ROOT_TO_BASKET } from './routes'
+import MainPage from './pages/MainPage';
 
 import './App.css';
-
 function App() {
   return (
-    <div className="app">
+    <div className="App">
```
Comment (Nerevand) [inline src/App.js]: better add `className` from lower case. 
And u don't have styles for class `.App`, but have for `.app` in file `App.css`

File: src/App.js [hunk 2]
```diff
@@ -1,53 +1,16 @@
...
-import { NavLink, Route, Routes } from "react-router-dom";
 
-import MainPage from "./pages/MainPage/MainPage";
-import Basket from "./pages/Basket";
-import NotFound from "./pages/Not found/Not found";
-import { ROOT_TO_MAINPAGE,ROOT_TO_BASKET } from './routes'
+import MainPage from './pages/MainPage';
 
 import './App.css';
-
```
Comment (Nerevand) [inline src/App.js:11]: should be empty line after  last import

File: src/Ui/Skeleton.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from "react"
```
Comment (Nerevand) [inline src/Ui/Skeleton.js]: u should create this Skeleton inside `components`

File: src/pages/Basket/Basket.module.scss [hunk 4]
```diff
@@ -55,51 +55,43 @@
...
+    display: inline-block;
+    font-family: Inter, -apple-system, system-ui, Roboto, "Helvetica Neue", Arial,
+      sans-serif;
+    height: 44px;
+    line-height: 44px;
+    outline: 0;
+    overflow: hidden;
+    padding: 0 20px;
+    pointer-events: auto;
+  //   position: absolute;
```
Comment (Nerevand) [inline src/pages/Basket/Basket.module.scss]: remove commented styles

File: src/pages/MainPage/MainPage.jsx [hunk 5]
```diff
@@ -1,24 +1,49 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx:1]: sort imports

File: src/pages/MainPage/MainPage.jsx [hunk 6]
```diff
@@ -1,24 +1,49 @@
...
 import Container from "react-bootstrap/Container";
 import Button from "react-bootstrap/Button";
 
-import { ADD_TO_BASKET } from "../../redux/store/actions";
+import Basket from "../Basket";
+
 import style from "./MainPage.module.scss";
+import { fetchUsers } from "../../services/Servise";
+import Skeleton from "../../Ui/Skeleton";
+const imgPerson =
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx:12]: after last import add empty line

File: src/pages/MainPage/MainPage.jsx [hunk 7]
```diff
@@ -1,24 +1,49 @@
...
+  "https://previews.123rf.com/images/fokaspokas/fokaspokas1806/fokaspokas180600645/103145234-team-few-person-white-icon-with-shadow-on-transparent-background.jpg";
 
 const MainPage = () => {
   const dispatch = useDispatch();
-  const products = useSelector((state) => state.items.it);
+  const users = useSelector((state) => state.items.users);
+  const [showSkeleton, setShowSkeleton] = useState(true);
+  const usersss = users[0];
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx]: `usersss` bad naming

File: src/pages/MainPage/MainPage.jsx [hunk 8]
```diff
@@ -1,24 +1,49 @@
...
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      dispatch(fetchUsers());
+    }, 2000);
+    return () => {
+      clearTimeout(timer);
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx:26]: why u clear timeout on unmont?

File: src/pages/MainPage/MainPage.jsx [hunk 9]
```diff
@@ -27,50 +52,61 @@ const MainPage = () => {
...
             <div className={style.box__list}>
-            {products.map((it, i) => {
-                const {name,img,price} = it;
-                return (
-                  <div key={i} className={style.box__card}>
-                    <div className={style.box__img}>
-                      <img src={img} alt="img_bags" />
+              {!usersss || showSkeleton ? (
+                <>
+                  {createSkeleton()}
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx]: create array on 4 items and `map` it

File: src/pages/MainPage/MainPage.jsx [hunk 10]
```diff
@@ -1,24 +1,49 @@
...
       data: Date.now(),
     };
-    dispatch({ type: ADD_TO_BASKET, payload: currentProduct });
+    dispatch({ type: "ADD_TO_BASKET", payload: currentProduct });
   };
+
+  function showBasket(cond) {
+    setShow(cond);
+  }
+  function createSkeleton() {
```
Comment (Nerevand) [inline src/pages/MainPage/MainPage.jsx]: u don't need this

File: src/redux/store/ItemsReduce.js [hunk 11]
```diff
@@ -1,8 +1,16 @@
 import imf from '../../img/bag_rose.jpg'
```
Comment (Nerevand) [inline src/redux/store/ItemsReduce.js:1]: a lot of empty lines. Remove it and lent only one. Don't need multiple empty lines

File: src/redux/store/index.js [hunk 12]
```diff
@@ -1,8 +1,9 @@
-import {createStore ,combineReducers} from 'redux'
-import { itemsReducer } from './ItemsReduce'
-
+import {createStore ,combineReducers,applyMiddleware} from 'redux'
+import { itemsReducer } from './ItemsReduce';
+import { composeWithDevTools } from 'redux-devtools-extension';
+import thunk from 'redux-thunk';
```
Comment (Nerevand) [inline src/redux/store/index.js:3]: add empty line after last import

Comment (Nerevand) [reply src/redux/store/index.js:3]: sort imports

File: src/services/Servise.js [hunk 13]
```diff
@@ -0,0 +1,9 @@
+import { addCustomersAction } from "../redux/store/ItemsReduce"
```
Comment (Nerevand) [inline src/services/Servise.js:1]: don't need to call that file from `UpperCase`
use naming with UpperCase only for `React Components`

File: src/pages/MainPage/MainPage.jsx (hunk 8 above)
Comment (Maximca2) [reply src/pages/MainPage/MainPage.jsx:26]: because i need to complete my timer working ,this is my mind!

**Review by Nerevand — APPROVED** [review]

**Pull request merged**